packages/lesswrong/components/posts/PostsAnalyticsPage.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import { Components, registerComponent } from '../../lib/vulcan-lib'
+import { useLocation, useServerRequestStatus } from '../../lib/routeUtil'
+import { useSingle } from '../../lib/crud/withSingle'
+import { useCurrentUser } from '../common/withUser'
+import { userOwns } from '../../lib/vulcan-users'
+import { usePostAnalytics } from './usePostAnalytics'
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableRow from '@material-ui/core/TableRow';
+import { Link } from '../../lib/reactRouterWrapper'
+import { forumTypeSetting } from '../../lib/instanceSettings'
+import NoSsr from '@material-ui/core/NoSsr';
+import classNames from 'classnames';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  viewingNotice: {
+    marginTop: theme.spacing.unit * 4,
+    '& a': {
+      color: theme.palette.primary.main,
+    },
+  },
+  // TODO: right now on lw-master, gutterBottom exists on the Typography class,
+  // but we don't yet have that code
+  gutterBottom: {
+    marginBottom: '0.35em',
+  },
+  calculating: {
+    marginLeft: theme.spacing.unit * 2,
+  },
+})
+
+const PostsAnalyticsInner = ({ classes, post }) => {
+  const { postAnalytics, loading, error } = usePostAnalytics(post._id)
+  const { Loading, Typography } = Components
+  if (loading) {
+    return <>
+      <Typography variant="body1" className={classNames(classes.gutterBottom, classes.calculating)}>
+        <em>Calculating metrics. (This can take some time for popular posts.)</em>
+      </Typography>
+      <Loading />
+    </>
+  }
+  if (error) {
+    throw error
+  }
+
+  return (<Table>
+    <TableBody>
+      <TableRow>
+        <TableCell>Views by unique devices</TableCell>
+        <TableCell>{postAnalytics?.uniqueClientViews}</TableCell>
+      </TableRow>
+      <TableRow>
+        <TableCell>{'Views by unique devices, on page for > 10 sec'}</TableCell>
+        <TableCell>{postAnalytics?.uniqueClientViews10Sec}</TableCell>
+      </TableRow>
+    </TableBody>
+  </Table>)
+}
+
+const PostsAnalyticsInnerComponent = registerComponent('PostsAnalyticsInner', PostsAnalyticsInner, {styles})
+
+declare global {
+  interface ComponentTypes {
+    PostsAnalyticsInner: typeof PostsAnalyticsInnerComponent
+  }
+}
+
+const PostsAnalyticsPage = ({ classes }) => {
+  const { query } = useLocation()
+  const { document: post } = useSingle({
+    documentId: query.postId,
+    collectionName: "Posts",
+    fragmentName: 'PostsPage',
+  })
+  const currentUser = useCurrentUser()
+  const serverRequestStatus = useServerRequestStatus()
+  const {
+    SingleColumnSection, WrappedLoginForm, PostsAnalyticsInner, HeadTags, Typography
+  } = Components
+
+  if (!query.postId) {
+    return <SingleColumnSection>
```

Comment:
Minor point: should this update the request status like the cases below it?

```suggestion
    if (serverRequestStatus) serverRequestStatus.status = 400
    return <SingleColumnSection>
```